### ISSUE TITLE
Example of <ul><template v-for><li> is invalid?

### ISSUE THREAD

**Issue by arty-name** [vuejs/vuejs.org#314]
The [list rendering](http://vuejs.org/guide/list.html) page provides following example for `<template v-for>`:

```html
<ul>
  <template v-for="item in items">
    <li>{{ item.msg }}</li>
    <li class="divider"></li>
  </template>
</ul>
```

At the same time [the components](http://vuejs.org/guide/components.html) page says: 

> Another consequence is that you can not use custom tags (including custom elements and special tags like `<component>`, `<template>` and `<partial>`) inside of `ul`, `select`, `table` and other elements with similar restrictions. Custom tags will be hoisted out and thus not render properly.

This is confusing. Is the example valid? If yes, how should one understand the text about special tags?

**Comment by simplesmiler** [comment]
The note about `<template>` is semi-valid.

1. By the standard `<template>` can appear anywhere, and can contain the tags that the template parent can contain, so the example is valid.
2. Some browsers don't quite follow the spec, and hoist the `<template>`. This is why we don't recommend to use `template` inside content-constraining tags.
3. By the standard `<ul>` can only contain `<li>`, but I don't know any browser that actually hoists non-`<li>` elements from inside the `<ul>`.

Note that Vue 2.0 will use custom parser instead of browser builtin one, so for non-DOM-based templates (`template: '<div></div>'`, `template: require('./index.html')` and single-file components) this quirks will not be an issue.

**Comment by simplesmiler** [comment]
I guess `ul` should be removed from the list in the note to adhere to the de facto behavior.

**Comment by chrisvfritz** [comment]
Just fixed this in f9461d0b3ab3e9baa112fffc0469cd1611e1495a.
